packages/react-components/react-positioning/src/utils/writeArrowUpdates.ts

```diff
@@ -1,18 +1,32 @@
-import { MiddlewareData } from '@floating-ui/dom';
+import { MiddlewareData, Placement } from '@floating-ui/dom';
 
 /**
  * Writes all DOM element updates after position is computed
  */
-export function writeArrowUpdates(options: { arrow: HTMLElement | null; middlewareData: MiddlewareData }) {
-  const { arrow, middlewareData } = options;
+export function writeArrowUpdates(options: {
+  arrow: HTMLElement | null;
+  middlewareData: MiddlewareData;
+  placement: Placement;
+}) {
+  const { arrow, middlewareData, placement } = options;
   if (!middlewareData.arrow || !arrow) {
     return;
   }
 
   const { x: arrowX, y: arrowY } = middlewareData.arrow;
 
+  const staticSide = {
+    top: 'bottom',
+    right: 'left',
+    bottom: 'top',
+    left: 'right',
+  }[placement.split('-')[0]] as string;
+
   Object.assign(arrow.style, {
-    left: `${arrowX}px`,
-    top: `${arrowY}px`,
+    left: arrowX !== null && arrowX !== undefined ? `${arrowX}px` : '',
+    top: arrowY !== null && arrowY !== undefined ? `${arrowY}px` : '',
+    right: '',
+    bottom: '',
+    [staticSide]: '-5px',
```

Comment:
<img width="894" alt="image" src="https://github.com/user-attachments/assets/ca1f2d36-fe4e-4efc-93ee-a41527f8bc10">

Something does not as expected :( Please check VR diffs.